spread.yaml

```diff
@@ -8,15 +8,43 @@ environment:
 backends:
   lxd:
     systems: [ubuntu-16.04]
+  linode:
+    key: "$(HOST: echo $SPREAD_LINODE_KEY)"
+    systems:
+      - ubuntu-16.04-64:
+          kernel: GRUB 2
+          workers: 3
 
 suites:
  spread_tests/:
    summary: tests of the snapcraft snap
    prepare: |
+     # apt update is hanging on security.ubuntu.com with IPv6, prefer IPv4 over IPv6
+     cat <<EOF > gai.conf
+     precedence  ::1/128       50
+     precedence  ::/0          40
+     precedence  2002::/16     30
+     precedence ::/96          20
+     precedence ::ffff:0:0/96 100
+     EOF
+     if ! mv gai.conf /etc/gai.conf; then
+       # not writable, disable ipv6 through sysctl
+       sysctl -w net.ipv6.conf.all.disable_ipv6=1
+       rm -f gai.conf
+     fi
+
      apt update
-     apt build-dep -y snapcraft
-     apt install -y bzr git curl mercurial subversion lxd python3-pep8 pyflakes python-flake8 python3-fixtures python3-pexpect python3-pyftpdlib python3-testscenarios
-     apt install -y squashfuse
+     apt install --yes gcc g++ make python3-dev python3-venv libffi-dev libsodium-dev libapt-pkg-dev libarchive13 squashfs-tools xdelta3 bzr git mercurial subversion
+     # Only required for the lxd backend.
+     apt install --yes squashfuse
+     mkdir -p /snapcraft/venv
+     python3 -m venv /snapcraft/venv
+     # XXX the activate script has an unbound variable.
+     sed -i '1s/^/set +u\n/' /snapcraft/venv/bin/activate
+     source /snapcraft/venv/bin/activate
+     pip install --upgrade pip
+     pip install -r /snapcraft/requirements.txt -r /snapcraft/requirements-devel.txt
+     apt install --yes snapd
      snap install snapcraft --classic --edge
```

Comment:
Wait... I don't understand what's happening here. You're installing via pip, but also installing the snap. They'll both be in the path. Which one is being tested?